tests/components/mqtt/test_discovery.py

```diff
@@ -550,6 +550,58 @@ def callback(event):
     assert events[3].data["old_state"] is None
 
 
+async def test_rapid_reconfigure(hass, mqtt_mock, caplog):
+    """Test immediate reconfigure of added component."""
+
+    events = []
+
+    @ha.callback
+    def callback(event):
+        """Verify event got called."""
+        events.append(event)
+
+    hass.bus.async_listen(EVENT_STATE_CHANGED, callback)
```

Comment:
We have this helper:
https://github.com/home-assistant/core/blob/44d8f2f7730f7fb6977e96d0e591323b5682e23e/tests/common.py#L1184